### PR TITLE
[DNM] msm8937-common: P bringup

### PR DESCRIPTION
**Pull request by JarlPenguin**
Apply device-specific commits (found here: https://github.com/moto-msm8937-common/android_device_motorola_montana/pull/1)

Do NOT merge until p-files.txt is done

### CODE REVIEW

**Comment by JarlPenguin** [comment]
+1, seems good to me, still needs testing

**Comment by JarlPenguin** [comment]
Cherry picked more commits from https://github.com/mvaisakh/platform_device_motorola_cedric

File: BoardConfigCommon.mk [hunk 1]
```diff
@@ -67,8 +65,8 @@ AUDIO_FEATURE_ENABLED_WMA_OFFLOAD := true
 AUDIO_USE_LL_AS_PRIMARY_OUTPUT := true
 AUDIO_FEATURE_ENABLED_SND_MONITOR := true
 BOARD_USES_ALSA_AUDIO := true
-BOARD_USES_GENERIC_AUDIO := true
-TARGET_USES_QCOM_MM_AUDIO := true
+USE_CUSTOM_AUDIO_POLICY := 1
+USE_XML_AUDIO_POLICY_CONF := 1
```
Comment (wiktorek140) [inline BoardConfigCommon.mk:70]: Should be 1? not true?

File: BoardConfigCommon.mk [hunk 2]
```diff
@@ -117,17 +116,20 @@ endif
...
+TARGET_QCOM_NO_FM_FIRMWARE := true
 
 # GPS
 TARGET_NO_RPC := true
 USE_DEVICE_SPECIFIC_GPS := true
 BOARD_VENDOR_QCOM_GPS_LOC_API_HARDWARE := $(TARGET_BOARD_PLATFORM)
 
 # Kernel
-BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 androidboot.hardware=qcom user_debug=30 msm_rtb.filter=0x237
+BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 androidboot.hardware=qcom user_debug=30 msm_rtb.filter=0x237  androidboot.usbconfigfs=false
```
Comment (wiktorek140) [inline BoardConfigCommon.mk]: additional space, probably it don't change anything

File: config.fs [hunk 3]
```diff
@@ -1,155 +1,161 @@
...
+[AID_VENDOR_QCOM_DIAG]
+value:2950
+
+[AID_VENDOR_RFS]
+value:2951
+
+[AID_VENDOR_RFS_SHARED]
+value:2952
+
+[system/vendor/bin/wcnss_filter]
```
Comment (wiktorek140) [inline config.fs]: Should be system/vendor/ or vendor/ ?

File: configs/privapp-permissions-hotword.xml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    </privapp-permissions>
+
+    <privapp-permissions package="com.android.hotwordenrollment.tgoogle">
+        <permission name="android.permission.MANAGE_VOICE_KEYPHRASES"/>
+    </privapp-permissions>
+
+    <privapp-permissions package="com.android.hotwordenrollment.xgoogle">
+        <permission name="android.permission.MANAGE_VOICE_KEYPHRASES"/>
+    </privapp-permissions>
+</permissions>
```
Comment (wiktorek140) [inline configs/privapp-permissions-hotword.xml:14]: It is even required?

File: gps/android/Gnss.h [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, The Linux Foundation. All rights reserved.
+ * Copyright (c) 2017-2018-2018-2018, The Linux Foundation. All rights reserved.
```
Comment (wiktorek140) [inline gps/android/Gnss.h]: why 2018 three times?

File: manifest.xml [hunk 6]
```diff
@@ -111,12 +124,13 @@
         <interface>
             <name>IGnss</name>
             <instance>default</instance>
+            <fqname>@1.0::IGnss/gnss_vendor</fqname>
```
Comment (wiktorek140) [inline manifest.xml]: 1.0 or 1.1 ?

File: msm8937.mk [hunk 7]
```diff
@@ -82,37 +82,48 @@ PRODUCT_PACKAGES += \
     libqcompostprocbundle
 
 PRODUCT_PACKAGES += \
-    android.hardware.audio@2.0-impl \
+    android.hardware.audio@4.0-impl \
     android.hardware.audio@2.0-service \
-    android.hardware.audio.effect@2.0-impl \
+    android.hardware.audio.effect@4.0-impl \
+    android.hardware.audio.effect@2.0-service \
```
Comment (wiktorek140) [inline msm8937.mk:54]: Why service 2.0?

File: msm8937.mk [hunk 8]
```diff
@@ -132,11 +143,18 @@ PRODUCT_PACKAGES += \
...
 
 # Display Calibration
 PRODUCT_PACKAGES += \
-    vendor.lineage.livedisplay@1.0-service-sdm
+    vendor.lineage.livedisplay@2.0-service-sdm \
+    vendor.lineage.livedisplay@2.0-service-sysfs
+
+#PRODUCT_COPY_FILES += \
+#    $(LOCAL_PATH)/configs/qdcm_calib_data_mipi_mot_vid_inx_1080p_497.xml:$(TARGET_COPY_OUT_VENDOR)/etc/qdcm_calib_data_mipi_mot_vid_inx_1080p_497.xml \
+#    $(LOCAL_PATH)/configs/qdcm_calib_data_mipi_mot_vid_tianma_1080p_497.xml:$(TARGET_COPY_OUT_VENDOR)/etc/qdcm_calib_data_mipi_mot_vid_tianma_1080p_497.xml
```
Comment (wiktorek140) [inline msm8937.mk]: Extract it to montana tree

File: msm8937.mk [hunk 9]
```diff
@@ -266,15 +293,20 @@ PRODUCT_PACKAGES += \
 # Permissions
 PRODUCT_COPY_FILES += \
     $(LOCAL_PATH)/configs/privapp-permissions-oem.xml:$(TARGET_COPY_OUT_SYSTEM)/etc/permissions/privapp-permissions-oem.xml \
+    $(LOCAL_PATH)/configs/privapp-permissions-hotword.xml:$(TARGET_COPY_OUT_SYSTEM)/etc/permissions/privapp-permissions-hotword.xml
```
Comment (wiktorek140) [inline msm8937.mk]: It is required?

File: BoardConfigCommon.mk (hunk 1 above)
Comment (JarlPenguin) [reply BoardConfigCommon.mk:70]: This is correct, since XML audio policy gets rid of those flags

File: msm8937.mk (hunk 7 above)
Comment (JarlPenguin) [reply msm8937.mk:54]: impl is 4.0, service is 2.0, idk why
but it's like that everywhere else, I'm assuming a 4.0-service doesn't exist

File: msm8937.mk (hunk 9 above)
Comment (JarlPenguin) [reply msm8937.mk]: fpushed out of branch

File: BoardConfigCommon.mk (hunk 2 above)
Comment (JarlPenguin) [reply BoardConfigCommon.mk]: done, corrected

File: msm8937.mk (hunk 8 above)
Comment (JarlPenguin) [reply msm8937.mk]: removed, we don't have these anyways

File: gps/android/Gnss.h (hunk 5 above)
Comment (JarlPenguin) [reply gps/android/Gnss.h]: done

File: manifest.xml (hunk 6 above)
Comment (JarlPenguin) [reply manifest.xml]: addressed, fqname should be 1.0

File: config.fs (hunk 3 above)
Comment (JarlPenguin) [reply config.fs]: fixed

File: rootdir/etc/ueventd.qcom.rc [hunk 10]
```diff
@@ -326,3 +327,8 @@
 
 #Egistec fingerprint
 /dev/esfp0		  0660   system     system
+
+# USB role switch
+/sys/class/dual_role_usb/*	data_role	0444	system	system
+/sys/class/dual_role_usb/*	power_role	0444	system	system
+/sys/class/dual_role_usb/*	mode		0444	system	system
```
Comment (wiktorek140) [inline rootdir/etc/ueventd.qcom.rc:395]: required for pie? or required for all devices?

File: sepolicy/file_contexts [hunk 11]
```diff
@@ -112,4 +107,14 @@
 
 /sys/kernel/debug/rmt_storage(/.*)? u:object_r:debugfs_rmt_storage:s0
 
-/data/vendor/nfc(/.*)? u:object_r:nfc_data_file:s0
+/(vendor|system/vendor)/bin/hw/android\.hardware\.health@2\.0-service\.msm8937        u:object_r:hal_health_default_exec:s0
+/(vendor|system/vendor)/bin/hw/android\.hardware\.vibrator@1\.0-service.msm8937   u:object_r:hal_vibrator_default_exec:s0
+/sys/devices/virtual/timed_output/vibrator/vtg_default                            u:object_r:sysfs_vibrator:s0
+/sys/devices/virtual/timed_output/vibrator/vtg_level                            u:object_r:sysfs_vibrator:s0
+/sys/devices/virtual/timed_output/vibrator/vtg_min                            u:object_r:sysfs_vibrator:s0
+/sys/devices/virtual/timed_output/vibrator/vtg_max                            u:object_r:sysfs_vibrator:s0
```
Comment (wiktorek140) [inline sepolicy/file_contexts]: Required for vibration?

File: system_prop.mk [hunk 12]
```diff
@@ -41,6 +41,7 @@ PRODUCT_PROPERTY_OVERRIDES += \
 PRODUCT_PROPERTY_OVERRIDES += \
     bluetooth.hfp.client=1 \
     qcom.bluetooth.soc=smd \
+    vendor.qcom.bluetooth.soc=smd \
```
Comment (wiktorek140) [inline system_prop.mk]: Looks like doubled. Should be both?

File: thermal/thermal_common.h [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+    THERM_SDM_439,
+    THERM_MSMNILE,
+    THERM_TALOS,
+    THERM_SDMMAGPIE,
+    THERM_MSM_8917,
+    THERM_TRINKET,
+    THERM_KONA,
+    THERM_LITO,
+    THERM_ATOLL,
+};
```
Comment (wiktorek140) [inline thermal/thermal_common.h:49]: I dont see MSM_8937. or we use 8953?

File: sepolicy/file_contexts (hunk 11 above)
Comment (JarlPenguin) [reply sepolicy/file_contexts]: for the new HAL it seems like it is required

File: system_prop.mk (hunk 12 above)
Comment (JarlPenguin) [reply system_prop.mk]: not doubled

File: thermal/thermal_common.h (hunk 13 above)
Comment (JarlPenguin) [reply thermal/thermal_common.h:49]: most likely sdm439, since it inherits from msm8937

**Comment by JarlPenguin** [comment]
-2, wifi dead

**Pull request closed without merging**